hnn_core/tests/test_params.py

```diff
@@ -74,29 +79,140 @@ def test_base_params():
+        outpath = Path(tmp_path, 'default.json')
+        convert_to_json(self.path_default,
+                        outpath
+                        )
+        net_json = read_network_configuration(outpath)
+        assert net_json == net_params
+
+        # Write json without drives
+        outpath_no_drives = Path(tmp_path, 'default_no_drives.json')
+        convert_to_json(self.path_default,
+                        outpath_no_drives,
+                        include_drives=False
+                        )
+        net_json_no_drives = read_network_configuration(outpath_no_drives)
+        assert net_json_no_drives != net_json
+        assert bool(net_json_no_drives.external_drives) is False
+
+        # Check that writing with no extension will add one
+        outpath_no_ext = Path(tmp_path, 'default_no_ext')
+        convert_to_json(self.path_default,
+                        outpath_no_ext
+                        )
+        assert outpath_no_ext.with_suffix('.json').exists()
+
+    def test_law_network_connectivity(self, tmp_path):
+        """Tests conversion with Law 2021 network connectivity model"""
+
+        net_params = law_2021_model(read_params(self.path_default),
+                                    add_drives_from_params=True,
+                                    )
+
+        # Write json and check if constructed network is equal
+        outpath = Path(tmp_path, 'default.json')
+        convert_to_json(self.path_default,
+                        outpath,
+                        network_connectivity='law_2021_model')
+        net_json = read_network_configuration(outpath)
+        assert net_json == net_params
+
+    def test_calcium_network_connectivity(self, tmp_path):
+        """Tests conversion with calcium network connectivity model"""
+
+        net_params = calcium_model(read_params(self.path_default),
+                                   add_drives_from_params=True,
+                                   )
+
+        # Write json and check if constructed network is equal
+        outpath = Path(tmp_path, 'default.json')
+        convert_to_json(self.path_default,
+                        outpath,
+                        network_connectivity='calcium_model')
+        net_json = read_network_configuration(outpath)
+        assert net_json == net_params
+
+    def test_no_network_connectivity(self, tmp_path):
+        """Tests conversion with no network connectivity model"""
+
+        net_params = Network(read_params(self.path_default),
+                             add_drives_from_params=True,
+                             )
+
+        # Write json and check if constructed network is equal
+        outpath = Path(tmp_path, 'default.json')
+        convert_to_json(self.path_default,
+                        outpath,
+                        network_connectivity=None)
+        net_json = read_network_configuration(outpath)
+        assert net_json == net_params
```

Comment:
```suggestion
        assert net_json == net_params
        assert len(net_json.connectivity) == len(net_param.connectivity) == 0
```

Reply:
I added this check but instead of 0, I'm using 22. This is because there are still 22 connections due to external drives.